handle_http.go

```diff
@@ -224,39 +224,77 @@ func download_http(source string, destination string, payload *payloadStruct, na
 		}
 	}
 
-	closestNamespaceCaches, err := GetCachesFromNamespace(namespace)
-	if err != nil {
-		log.Errorln("Failed to get namespaced caches (treated as non-fatal):", err)
-	}
-	log.Debugln("Matched caches:", closestNamespaceCaches)
-
-	// Make sure we only try as many caches as we have
-	cachesToTry := CachesToTry
-	if cachesToTry > len(closestNamespaceCaches) {
-		cachesToTry = len(closestNamespaceCaches)
-	}
-	log.Debugln("Trying the caches:", closestNamespaceCaches[:cachesToTry])
+	// Check the env var "USE_OSDF_DIRECTOR" and decide if ordered caches should come from director
+	//var closestNamespaceCaches []Cache
 	var transfers []TransferDetails
-	downloadUrl := url.URL{Path: source}
 	var files []string
+	if OSDFDirectorUrl != "" {
```

Comment:
I encourage you to think deeply about how you can restructure this large if statement.  I feel both inside the if and else are doing very similar things, just using different data sources for the caches.  I think with 2 if's instead of 1 large one, you can reduce code duplication significantly.